### PR TITLE
common unit tests for xfpga

### PR DESCRIPTION
**Pull request by jelonanX**

### CODE REVIEW

File: testing/xfpga/test_common_c.cpp [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+// IMPLIED WARRANTIES OF  MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED.  IN NO EVENT  SHALL THE COPYRIGHT OWNER  OR CONTRIBUTORS BE
+// LIABLE  FOR  ANY  DIRECT,  INDIRECT,  INCIDENTAL,  SPECIAL,  EXEMPLARY,  OR
+// CONSEQUENTIAL  DAMAGES  (INCLUDING,  BUT  NOT LIMITED  TO,  PROCUREMENT  OF
+// SUBSTITUTE GOODS OR SERVICES;  LOSS OF USE,  DATA, OR PROFITS;  OR BUSINESS
+// INTERRUPTION)  HOWEVER CAUSED  AND ON ANY THEORY  OF LIABILITY,  WHETHER IN
+// CONTRACT,  STRICT LIABILITY,  OR TORT  (INCLUDING NEGLIGENCE  OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,  EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+#ifdef __cplusplus
```
Comment (tswhison) [inline testing/xfpga/test_common_c.cpp]: Remove.. :)

File: testing/xfpga/test_common_c.cpp [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+using namespace opae::testing;
+
+/**
+ * @test       common_01
+ *
+ * @brief      Verifies the string returned by fpgaErrStr() for each
+ *             fpga_result enumeration value.
+ */
+TEST(LibopaecErrorCommonALL, common_01) {
```
Comment (r-rojo) [inline testing/xfpga/test_common_c.cpp]: can we rename this to
```C++
TEST(common, fpgaErrStr)
```

File: testing/xfpga/test_common_c.cpp [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+  virtual void SetUp() override {
+    ASSERT_TRUE(test_platform::exists(GetParam()));
+    platform_ = test_platform::get(GetParam());
+    system_ = test_system::instance();
+    system_->initialize();
+    tmpsysfs_ = system_->prepare_syfs(platform_);
+
+    ASSERT_EQ(xfpga_fpgaGetProperties(nullptr, &filter_), FPGA_OK);
+    //ASSERT_EQ(fpgaPropertiesSetObjectType(filter_, FPGA_ACCELERATOR), FPGA_OK);
+    ASSERT_EQ(xfpga_fpgaPropertiesSetObjectType(filter_, FPGA_ACCELERATOR), FPGA_OK);
```
Comment (tswhison) [inline testing/xfpga/test_common_c.cpp]: This API no longer exists.  Replace line 60 with (uncommented) line 59 to fix.

File: testing/xfpga/test_common_c.cpp [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+    ASSERT_EQ(xfpga_fpgaEnumerate(&filter_, 1, tokens_.data(), tokens_.size(),
+                            &num_matches_),
+              FPGA_OK);
+    ASSERT_EQ(xfpga_fpgaOpen(tokens_[0], &handle_, 0), FPGA_OK);
+    ASSERT_EQ(xfpga_fpgaCreateEventHandle(&eh_), FPGA_OK);
+  }
+
+  virtual void TearDown() override {
+    //EXPECT_EQ(fpgaDestroyProperties(&filter_), FPGA_OK);
+    EXPECT_EQ(xfpga_fpgaDestroyProperties(&filter_), FPGA_OK);
```
Comment (tswhison) [inline testing/xfpga/test_common_c.cpp]: This API no longer exists.  Replace line 70 with (uncommented) line 69 to fix.

File: testing/CMakeLists.txt [hunk 5]
```diff
@@ -110,6 +110,10 @@ add_mock_test(test_xfpga_token_list_c xfpga-static
     xfpga/test_token_list_c.cpp
 )
 
+add_mock_test(test_xfpga_common_c xfpga-static
+    xfpga/test_common_c.cpp
+    ${OPAE_SDK_SOURCE}/libopae/plugins/xfpga/event.c
```
Comment (tswhison) [inline testing/CMakeLists.txt]: Not sure if you intended to pull event.c into this test?

File: testing/xfpga/test_common_c.cpp [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+    system_->initialize();
+    tmpsysfs_ = system_->prepare_syfs(platform_);
+
+    ASSERT_EQ(xfpga_fpgaGetProperties(nullptr, &filter_), FPGA_OK);
+    ASSERT_EQ(fpgaPropertiesSetObjectType(filter_, FPGA_ACCELERATOR), FPGA_OK);
+    ASSERT_EQ(xfpga_fpgaEnumerate(&filter_, 1, tokens_.data(), tokens_.size(),
+                            &num_matches_),
+              FPGA_OK);
+    ASSERT_EQ(xfpga_fpgaOpen(tokens_[0], &handle_, 0), FPGA_OK);
+    ASSERT_EQ(xfpga_fpgaCreateEventHandle(&eh_), FPGA_OK);
```
Comment (tswhison) [inline testing/xfpga/test_common_c.cpp]: Just a nit, the tests aren't currently using eh_.  Was this a copy/paste remnant from the event tests?

File: testing/xfpga/test_common_c.cpp [hunk 7]
```diff
@@ -56,7 +57,7 @@ class common_c_p
     tmpsysfs_ = system_->prepare_syfs(platform_);
 
     ASSERT_EQ(xfpga_fpgaGetProperties(nullptr, &filter_), FPGA_OK);
-    ASSERT_EQ(fpgaPropertiesSetObjectType(filter_, FPGA_ACCELERATOR), FPGA_OK);
+    ASSERT_EQ(fpgaPropertiesSetObjectType(filter_, FPGA_DEVICE), FPGA_OK);
     ASSERT_EQ(xfpga_fpgaEnumerate(&filter_, 1, tokens_.data(), tokens_.size(),
```
Comment (tswhison) [inline testing/xfpga/test_common_c.cpp]: Please be sure to called fpgaDestroyToken() for each tokens_ in TearDown().

File: testing/CMakeLists.txt (hunk 5 above)
Comment (tswhison) [reply testing/CMakeLists.txt]: ok

**Pull request merged**